tinygrad/runtime/ops_python.py

```diff
@@ -120,7 +120,10 @@ def __call__(self, *bufs, global_size:Tuple[int,int,int]=(1,1,1), local_size:Tup
             ul[i] = inp
           else:
             # TODO: add real cast
-            if dtypes.is_int(dtype):
+            if isinstance(arg, tuple) and arg[1] or (dtypes.is_int(dtype) and dtypes.is_int(dtp[0]) and dtype.itemsize == dtp[0].itemsize):
```

Comment:
What is this second part of this if clause with the ints?

Reply:
Bitcasting between int and uint of the same size.

Reply:
Huh? If it's bitcast, wouldn't the first clause get it?

Reply:
Casting between int and uint and back in C has the same behavior as bitcasting, but isn't considered bitcasting, thus first part of clause doesn't get it. But doing proper cast would be better for sure.